docs/source/development/documentation.md

```diff
@@ -0,0 +1,110 @@
+change whenever the notebook is run, not just when there are meaningful changes to the
+content.
+
+When the package documentation is built in `sphinx` the `myst_nb` extension
+automatically runs any code in MyST markdown documents and renders the resulting code
+and outputs into the HTLM. The documentation can therefore contain dynamically generated
+examples of running code and output plots to demonstrate how to use the
+package. This build process also ensures that code in documentation is kept up to date
+with the package. The syntax for including code blocks in documents is shown below -
+this example includes a tag that causes the code to be concealed in the built
+documentation, but with a toggle button to allow it to be expanded.
+
+````{code-block}
+```{code-cell} python
+:tags: [hide-input]
+# This is just an example code cell to demonstrate how code is included in
+# the pyrealm documentation.
+```
+````
+
+### Docstrings
+
+The `pyrealm` package uses docstrings written in the [Google
+style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html).
+This allows the function documentation to be stored alongside the code and it is included
+in the documentation using the `sphinx` `autodoc` extension. See the code itself for
+examples of the documentation formatting and typical content.
+
+Unlike the files in `docs/sources` pages, the docstrings in code need to be written
+using reStructuredText: at present the `autodoc` functions in `sphinx` currently rely on
+`rst` inputs.
+
+### Referencing
+
+Both the `docs/source` and docstrings uses the `sphinxcontrib-bibtex` package to support
+citations. This uses Latex like citation keys in the documentation to insert references
+and build a bibliography. The `sphinx` configuration in `docs/source/conf.py` provides a
+custom Author/Year citation style. The reference library in `source/refs.bib` needs to
+be kept up to date with the literature for the project.
+
+## Building the documentation
+
+The `sphinx` package is used to build an HTML version of the package documentation
+provided in `docs/source` and to include the API documentation provided in the code
+docstrings. The `sphinx` building process requires some extra packages, but these are
+included in the `docs` group in `pyproject.toml` and should be installed.
+
+In order to build the package documentation, Jupyter needs to be able to associate the
+documentation files with the Python environment managed by `poetry`. This is done by
+installing the `poetry` environment as a new Jupyter kernel with a fixed name. This
+allows all build systems to run notebooks using the correct build environment:
+
+```bash
+poetry run python -m ipykernel install --user --name=pyrealm_python3
+```
+
+In order to build the package documentation, the following command can then be used:
+
+```bash
+# Build docs using sphinx
+cd docs
+poetry run sphinx-build -W --keep-going source build
+```
```

Comment:
Are there any commands that need to be put at the top of a file to be picked up in the documentation?